brave/src/main/java/zipkin2/reporter/brave/ZipkinSpanHandler.java

```diff
@@ -0,0 +1,203 @@
+     *
+     * @since 2.13
+     */
+    public Builder errorParser(ErrorParser errorParser) {
+      this.errorParser = errorParser;
+      return this;
+    }
+
+    /**
+     * When true, all spans {@link TraceContext#sampledLocal() sampled locally} are reported to the
+     * span reporter, even if they aren't sampled remotely. Defaults to {@code false}.
+     *
+     * <p>The primary use case is to implement a <a href="https://github.com/openzipkin-contrib/zipkin-secondary-sampling">sampling
+     * overlay</a>, such as boosting the sample rate for a subset of the network depending on the
+     * value of a baggage field. This means that data will report when either the trace is normally
+     * sampled, or secondarily sampled via a custom header.
+     *
+     * <p>This is simpler than a custom {@link FinishedSpanHandler}, because you don't have to
+     * duplicate transport mechanics already implemented in the {@link Reporter span reporter}.
+     * However, this assumes your backend can properly process the partial traces implied when using
+     * conditional sampling. For example, if your sampling condition is not consistent on a call
+     * tree, the resulting data could appear broken.
+     *
+     * @see TraceContext#sampledLocal()
+     * @since 2.13
+     */
+    public Builder alwaysReportSpans() {
+      this.alwaysReportSpans = true;
+      return this;
+    }
+
+    public FinishedSpanHandler build() {
+      if (spanReporter == Reporter.NOOP) return FinishedSpanHandler.NOOP;
+      return new ZipkinSpanHandler(this);
+    }
+  }
+
+  final Reporter<Span> spanReporter;
+  final ErrorParser errorParser;
+  final boolean alwaysReportSpans;
+
+  ZipkinSpanHandler(Builder builder) {
+    this.spanReporter = builder.spanReporter;
+    this.errorParser = builder.errorParser;
+    this.alwaysReportSpans = builder.alwaysReportSpans;
+  }
+
+  @Override public boolean handle(TraceContext context, MutableSpan span) {
+    if (!alwaysReportSpans && !Boolean.TRUE.equals(context.sampled())) return true;
+    maybeAddErrorTag(span);
+    Span converted = convert(context, span);
+    spanReporter.report(converted);
+    return true;
+  }
+
+  // Do not use @Override annotation to ensure compatibility with Brave 5.6
+  public boolean supportsOrphans() {
+    return true;
+  }
+
+  static Span convert(TraceContext context, MutableSpan span) {
+    Span.Builder result = Span.newBuilder()
+      .traceId(context.traceIdString())
+      .parentId(context.parentIdString())
+      .id(context.spanId())
+      .name(span.name());
+
+    long start = span.startTimestamp(), finish = span.finishTimestamp();
+    result.timestamp(start);
+    if (start != 0 && finish != 0L) result.duration(Math.max(finish - start, 1));
+
+    // use ordinal comparison to defend against version skew
+    brave.Span.Kind kind = span.kind();
+    if (kind != null && kind.ordinal() < Span.Kind.values().length) {
```

Comment:
Just wondering do we have a unit test that checks for ordinal drift?

Reply:
doubt it, and hope we never add any kinds, but this does deserve a unit test. it would have to be quite a thing (like powermock?) to change an enum's ordinal to make this fail.

Reply:
Was just thinking of one that checks that the enum items with same name have same ordinal. But true kind change seems unlikely enough that not a big deal

Reply:
probably a switch statement is easier than powermock.

This was originally here to allow producer/consumer kind which didn't exist in one or the other at the time (I don't recall). I don't expect us to add any more kinds and someone can do powermock later if we do :)

Reply:
I changed to a static initialized map, as I think it is overall easier to comprehend and  cleaner. thanks for noticing